src/server/project.ts

```diff
@@ -2318,15 +2318,15 @@ export abstract class Project implements LanguageServiceHost, ModuleResolutionHo
     runWithTemporaryFileUpdate(rootFile: string, updatedText: string, cb: (updatedProgram: Program, originalProgram: Program | undefined, updatedFile: SourceFile) => void) {
         const originalProgram = this.program;
         const rootSourceFile = Debug.checkDefined(this.program?.getSourceFile(rootFile), "Expected file to be part of program");
-        const originalText = Debug.checkDefined(rootSourceFile.getText());
+        const originalText = Debug.checkDefined(rootSourceFile.getFullText());
 
         this.getScriptInfo(rootFile)?.editContent(0, originalText.length, updatedText);
         this.updateGraph();
         try {
             cb(this.program!, originalProgram, (this.program?.getSourceFile(rootFile))!);
         }
         finally {
-            this.getScriptInfo(rootFile)?.editContent(0, this.program!.getSourceFile(rootFile)!.getText().length, originalText);
+            this.getScriptInfo(rootFile)?.editContent(0, updatedText.length, originalText);

```

Comment:
Hmm... looking at it again, is it really safe to assume that `cb` doesn't itself change the contents of the file? I'm sure the way we use this function is used in our codebase, it's fine, but I just want to check. What do you think @sheetalkamat?

Reply:
yes we should assume cb doesnt change the contents of file since file is never "muted" without change in script info

Reply:
(assuming you mean mutated)

What I meant was that `cb` could itself run `editContent` somehow. If that happens, the `ScriptInfo` object may not actually contain the same text anymore.

I'm okay with assuming that for now, I just wanted to run it by you.

Reply:
The alternative is just to replace the text contents with the script info's own text contents. It should all be available anyway.

Reply:
Yes I did mean mutated ☺️ Currently editing script info from callback is not possible as it’s just trying to calculate import fixes.. when that changes we can revisit this  ..